### PR TITLE
docs(autocomplete): add autocomplete docs

### PR DESCRIPTION
**Pull request by kara**
Docs site example to come once examples live here.

r: @jelbourn @andrewseguin

### CODE REVIEW

File: src/lib/autocomplete/OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      {{ option }}
+   </md-option>
+</md-autocomplete>
+```
+
+*my-comp.ts*
+```ts
+class MyComp {
+   myControl = new FormControl();
+   options = [
```
Comment (jelbourn) [inline src/lib/autocomplete/OVERVIEW.md]: Use a real type here and get rid of `any` throughout? Maybe something like `new User('Mary', 'Shelly')`

File: src/lib/autocomplete/OVERVIEW.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+   </md-option>
+</md-autocomplete>
+```
+
+### Adding a custom filter
+
+At this point, the autocomplete panel should be toggleable on focus and options should be selectable. But if we want 
+our options to filter when we type, we need to add a custom filter. 
+
+We already have access to the built-in `valueChanges` observable on the `FormControl`, so we can simply map the text
```
Comment (jelbourn) [inline src/lib/autocomplete/OVERVIEW.md]: This feels a bit like you're saying "This is the One True Way to do filtering". What do you think of approaching more as "You can filter the options in any way you want based on the text input. Here's one example."

File: src/lib/autocomplete/OVERVIEW.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+### Setting separate control and display values
+
+If you want the option's control value (what is saved in the form) to be different than the option's display value 
+(what is displayed in the actual text field), you'll need to set the `displayWith` property on your autocomplete 
+element. A common use case for this might be if you want to save your data as an object, but display just one of 
+the option's string properties.
+
+To make this work, create a function on your component class that maps the control value to the desired display value. 
+Then bind it to the autocomplete's `displayWith` property. You'll probably also want to add a step to your
```
Comment (jelbourn) [inline src/lib/autocomplete/OVERVIEW.md]: `You probably also want to` => `You can`

File: src/lib/autocomplete/OVERVIEW.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+         .map(val => this.displayFn(val))
+         .map(name => name ? this.filter(name) : this.options.slice());
+   }
+   
+   filter(val: string): any[] {
+      return this.options.filter(option => option.match(new RegExp(val, 'gi'))); 
+   }
+   
+   displayFn(value: any): string {
+      return value && typeof value === 'object' ? value.name : value;
```
Comment (jelbourn) [inline src/lib/autocomplete/OVERVIEW.md]: Why is `typeof` needed here?

File: src/lib/autocomplete/OVERVIEW.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+   ngOnInit() { 
+      this.filteredOptions = this.myControl.valueChanges
+         .startWith(null)
+         .map(val => this.displayFn(val))
+         .map(name => name ? this.filter(name) : this.options.slice());
+   }
+   
+   filter(val: string): any[] {
+      return this.options.filter(option => option.match(new RegExp(val, 'gi')));
```
Comment (jelbourn) [inline src/lib/autocomplete/OVERVIEW.md]: You typically don't want to use `match` when you are looking for a yes/no result. 
http://stackoverflow.com/a/10940138/477102

**Comment by kara** [comment]
@jelbourn Comments addressed

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**